query.go

```diff
@@ -438,6 +438,12 @@ func (c *Conn) Query(sql string, args ...interface{}) (*Rows, error) {
 		}
 	}
 
+	// Set query execution deadline
+	if c.config.QueryExecTimeout > 0 {
+		c.conn.SetDeadline(time.Now().Add(c.config.QueryExecTimeout))
+		defer c.conn.SetDeadline(time.Time{}) // reset deadline
```

Comment:
maybe this could be done in the Release() method? It seems like these defer statements will modify `c.conn` after it's already been released by this method